### PR TITLE
singer.read_tap - read from a tap programatically

### PR DESCRIPTION
**Pull request by OriHoch**
Adds a `singer.read_tap` method which starts a tap using subprocess.Propen and yields Message objects from it

Doesn't handle errors / exceptions

See read_tap function documentation and tests for more details.

related issue: #84

### CODE REVIEW

File: tests/bin/tap-mockexchangerates [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env sh
```
Comment (dmosorast) [inline tests/bin/tap-mockexchangerates]: This is a neat solution to running the test, but adding a dependency on `sh` seems like a bit much.

I'd like to keep this all in Python as much as possible, which, in this case can be accomplished by extracting the Popen call to a function that returns `tap`, and, in the test, replacing it with a mock function that just returns an object with the `stdout` property present that will provide the iterating interface expected by the rest of the code.

Does that make sense?

File: singer/messages.py [hunk 2]
```diff
@@ -276,3 +278,26 @@ def write_version(stream_name, version):
     write_version(stream, version)
     """
     write_message(ActivateVersionMessage(stream_name, version))
+
+
+def read_tap(name, config, **kwargs):
```
Comment (dmosorast) [inline singer/messages.py]: I notice that this only works for taps that only take a config, and doesn't support the rest of the standard parameters to taps. Could this be extended to include `--state` and `--catalog`?

`--discover` is a bit different, but should be captured in this style of interface to maximize usefulness. Adding a comment to the whole review on the topic.

**Comment by OriHoch** [comment]
thanks @dmosorast your comments make sense :+1: I'm just not sure when I'll get around to implement the changes.. hopefully in a week or two

**Comment by timvisher** [comment]
@OriHoch Think you'll have any time to work on this in the next week or so? If not it's NBD we can close the PR/Issue out for now and when you have time to pick it back up we can get it going again.

Thoughts?

**Comment by timvisher** [comment]
We'll close this out so we get re-notified if work picks back up on it. :)

**Pull request closed without merging**